pkg/operator/ceph/object/health.go

```diff
@@ -122,39 +133,28 @@ func (c *bucketChecker) checkObjectStoreHealth() error {
 
 		Always keep the bucket and the user for the health check, just do PUT and GET because bucket creation is expensive
 	*/
-	var tlsCert []byte
-	var s3AccessKey string
-	var s3SecretKey string
-	var err error
-	s3endpoint := BuildDNSEndpoint(BuildDomainName(c.objContext.Name, c.namespacedName.Namespace), c.port, c.objectStoreSpec.IsTLSEnabled())
-
-	// Generate unique user and bucket name
-	bucketName := genUniqueBucketName(c.objContext.UID)
-	userConfig := c.genUserConfig()
 
-	var httpClient *http.Client
-	if c.objectStoreSpec.IsTLSEnabled() {
-		tlsCert, err = GetTlsCaCert(c.objContext, c.objectStoreSpec)
-		if err != nil {
-			return errors.Wrapf(err, "failed to fetch CA cert for the user %q to establish TLS connection with the object store %q", userConfig.ID, c.namespacedName.Name)
-		}
-		httpClient.Transport = BuildTransportTLS(tlsCert)
+	// Keep admin ops context up-to date if there are config changes
+	if err := UpdateContext(&c.objContext.Context, c.objectStoreSpec); err != nil {
+		return errors.Wrapf(err, "failed to parse updated CephObjectStore spec")
 	}
-
-	// Build admin Ops API connection
-	co, err := admin.New(s3endpoint, c.objContext.adminOpsUserAccessKey, c.objContext.adminOpsUserSecretKey, httpClient)
+	opsCtx, err := NewMultisiteAdminOpsContext(&c.objContext.Context, c.objectStoreSpec)
 	if err != nil {
 		return errors.Wrap(err, "failed to build admin ops API connection")
 	}
-	c.objContext.adminOpsClient = co
+	c.objContext = opsCtx
+
```

Comment:
It took me a while to figure out how to create more reuse from this function to creating the admin api for the dependency checker. I moved most of this into the `NewMultisiteAdminOpsConfig()` function, but I was confused why this is done in this method as opposed to front-loading the work when the health checker object is created.

My best guess as to why this was done in this function was to support the case where an object store's spec is updated and the health checker needs to keep working. Thus the `UpdateEndpoint()` function and setting `c.objContext = opsCtx` with each run here.

Does this sound right @leseb ?